### PR TITLE
adding list invoice function

### PR DESCRIPTION
**Pull request by KAVAN-DESAI**

### CODE REVIEW

**Review by vincenzopalazzo — COMMENTED** [review]
Can you add a test? :)

**Review by vincenzopalazzo — COMMENTED** [review]
Yeah the build directory should be removed!

File: lib/src/cln_grpc_base.dart [hunk 1]
```diff
@@ -122,6 +122,16 @@ class GRPCClient extends LightningClient {
     return response as T;
   }
 
+  Future<T> listInvoices<T>(
+      {required ListinvoicesRequest params, T Function(Map)? onDecode}) async {
+    /// request to server
+    var response = await stub.listInvoices(params);
+    if (onDecode != null) {
+      return onDecode(toEncode(response.toProto3Json()));
```
Comment (vincenzopalazzo) [inline lib/src/cln_grpc_base.dart:130]:
```suggestion
      return onDecode(response.toProto3Json());
```

You are passing already a JSON

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
We need to call jsonEncode? I think no

File: lib/src/cln_grpc_base.dart (hunk 1 above)
Comment (KAVAN-DESAI) [reply lib/src/cln_grpc_base.dart:130]: can we directly do `return onDecode(response.writeToJsonMap());`?

Comment (vincenzopalazzo) [reply lib/src/cln_grpc_base.dart:130]: I think you can try, and if the return type is an object we should be able to make a cast to `Map<String, dynamic>`

like `response as Map<String, dynamic>`

I think you can try this and open a new PR to see if the test are happy or now?

**Review by vincenzopalazzo — COMMENTED** [review]
if you remove the build dir I can merge this, and you can make the cast experiment in another PR to be sure that you ran the test and see if I'm wrong with the suggestion.

**Review by vincenzopalazzo — APPROVED** [review]
Landing at the end of the CI check

**Pull request merged**